course/page/code.py

```diff
@@ -613,8 +613,7 @@ def transfer_attr(name):
         transfer_attr("names_for_user")
         transfer_attr("names_from_user")
 
-        if hasattr(self.page_desc, "test_code"):
-            run_req["test_code"] = self.get_test_code()
+        run_req["test_code"] = self.get_test_code()
```

Comment:
`if hasattr(self.page_desc, "test_code"):` is done in `self.get_test_code()`.

Reply:
LGTM